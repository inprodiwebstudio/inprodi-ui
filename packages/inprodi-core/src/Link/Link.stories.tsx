import React             from "react";
import { BrowserRouter } from "react-router-dom";

import { ComponentMeta, ComponentStory } from "@storybook/react";
import { Link, LinkProps }               from "./Link";

export default {
	title     : "Core/Typography/Link",
	component : Link,
	argTypes  : {
		color : { control : "color" },
	},
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = (args : LinkProps) => {

	return(
		<BrowserRouter>
			<Link {...args} >
			    Hola
			</Link>
		</BrowserRouter>
	);
};

export const Basic = Template.bind({});
Basic.args = {
	to : "/",
};
