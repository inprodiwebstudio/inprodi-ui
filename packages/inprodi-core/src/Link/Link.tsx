import { Text, TextProps }                  from "@mantine/core";
import { RelativeRoutingType, useNavigate } from "react-router-dom";

export interface LinkProps extends TextProps {
    to : string;
    state ?: any;
    replace ?: boolean;
    preventScrollReset ?: boolean;
    relative ?: RelativeRoutingType;
    onClick ?: React.MouseEventHandler<HTMLDivElement>
}

export const Link = ({
	to,
	state,
	replace,
	children,
	relative,
	preventScrollReset,
	onClick,
	color = "blue",
	...rest
} : LinkProps) => {
	const navigate  = useNavigate();
	const handleOnClick = (e : React.MouseEvent<HTMLDivElement, MouseEvent>) => {
		navigate(to, { relative, replace, preventScrollReset, state });
		onClick && onClick(e);
	};
	return (
		<Text
			{...rest}
			color={color}
			onClick={handleOnClick}
			style={{
				cursor         : "pointer",
				textDecoration : "underline",
				width          : "fit-content",
			}}
		>
			{children}
		</Text>
	);
};

export default Link;
