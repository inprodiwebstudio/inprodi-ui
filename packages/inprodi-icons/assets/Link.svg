<svg width="14" height="15" viewBox="0 0 14 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.2299 0.599976C10.5187 0.599976 9.80762 0.869419 9.2684 1.40864L7.29478 3.38226C6.36177 4.31526 6.28587 5.73946 6.96723 6.81133L6.21152 7.56704C5.13964 6.8856 3.71549 6.9608 2.78245 7.89383L0.808829 9.86745C-0.26961 10.9459 -0.26961 12.7128 0.808829 13.7912C1.88754 14.8695 3.65425 14.8696 4.73262 13.7912L6.70625 11.8176C7.63928 10.8846 7.71447 9.46044 7.03304 8.38856L7.78875 7.63285C8.86073 8.31408 10.2849 8.23825 11.2178 7.3053L13.1914 5.33168C14.2699 4.25324 14.2699 2.48708 13.1914 1.40864C12.6522 0.869419 11.9411 0.599976 11.2299 0.599976ZM11.2299 1.75434C11.6413 1.75434 12.0528 1.91304 12.3699 2.23016C13.0042 2.86441 13.0042 3.8759 12.3699 4.51015L10.3963 6.48378C9.90804 6.97203 9.2 7.0763 8.60876 6.81284L9.87811 5.54349C9.93387 5.48995 9.97838 5.42583 10.009 5.35488C10.0397 5.28393 10.0559 5.20757 10.0567 5.13028C10.0575 5.05299 10.0428 4.97632 10.0136 4.90476C9.98441 4.8332 9.94121 4.76818 9.88655 4.71353C9.83189 4.65887 9.76688 4.61567 9.69532 4.58645C9.62376 4.55724 9.54709 4.54259 9.4698 4.54338C9.39251 4.54416 9.31615 4.56036 9.2452 4.59103C9.17424 4.62169 9.11012 4.66621 9.05659 4.72196L7.78724 5.99132C7.52386 5.40018 7.62809 4.69199 8.1163 4.20378L10.0899 2.23016C10.407 1.91304 10.8185 1.75434 11.2299 1.75434ZM4.74473 8.24029C4.96649 8.24029 5.18451 8.29491 5.39151 8.38705L4.12291 9.65564C4.06716 9.70918 4.02264 9.7733 3.99198 9.84425C3.96131 9.9152 3.94511 9.99156 3.94433 10.0689C3.94354 10.1461 3.95818 10.2228 3.9874 10.2944C4.01662 10.3659 4.05982 10.4309 4.11447 10.4856C4.16913 10.5403 4.23414 10.5835 4.3057 10.6127C4.37726 10.6419 4.45394 10.6565 4.53123 10.6558C4.60852 10.655 4.68488 10.6388 4.75583 10.6081C4.82678 10.5774 4.8909 10.5329 4.94443 10.4772L6.21303 9.20857C6.4761 9.7996 6.37278 10.5073 5.88472 10.9953L3.91034 12.9697C3.27603 13.604 2.2651 13.6035 1.63035 12.969C0.996567 12.3347 0.996256 11.3238 1.63035 10.6897L3.60473 8.71535C3.92186 8.39823 4.3333 8.24029 4.74473 8.24029Z" fill="black"/>
</svg>
