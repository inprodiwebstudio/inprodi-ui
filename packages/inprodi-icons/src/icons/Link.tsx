import * as React from "react";
import { SVGProps } from "react";
const SvgLink = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width="1em"
    height="1em"
    viewBox="0 0 14 15"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M11.23.6c-.711 0-1.422.27-1.962.809L7.295 3.382c-.933.933-1.01 2.357-.328 3.43l-.755.755c-1.072-.681-2.497-.606-3.43.327L.81 9.867a2.784 2.784 0 0 0 0 3.924 2.784 2.784 0 0 0 3.924 0l1.973-1.973c.933-.933 1.008-2.358.327-3.43l.756-.755c1.072.681 2.496.605 3.429-.328l1.973-1.973a2.783 2.783 0 0 0 0-3.923A2.765 2.765 0 0 0 11.23.6Zm0 1.154c.411 0 .823.159 1.14.476a1.604 1.604 0 0 1 0 2.28l-1.974 1.974a1.595 1.595 0 0 1-1.787.329l1.27-1.27a.58.58 0 1 0-.822-.821l-1.27 1.27c-.263-.592-.159-1.3.33-1.788L10.09 2.23a1.607 1.607 0 0 1 1.14-.476ZM4.745 8.24c.221 0 .44.055.647.147l-1.27 1.269a.581.581 0 1 0 .822.821L6.213 9.21c.263.59.16 1.298-.328 1.786L3.91 12.97a1.604 1.604 0 0 1-2.28-.001 1.604 1.604 0 0 1 0-2.28l1.975-1.974a1.606 1.606 0 0 1 1.14-.475Z"
      fill="currentColor"
    />
  </svg>
);
export default SvgLink;
