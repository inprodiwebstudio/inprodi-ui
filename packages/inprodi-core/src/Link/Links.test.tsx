import React                      from "react";
import { act, fireEvent, render } from "@testing-library/react";
import { BrowserRouter }          from "react-router-dom";
import { Link }                   from "./Link";

describe("Links", () => {
	test("snapshot", () => {
		const { container } = render(
			<BrowserRouter>
				<Link to="/">Path Label</Link>
			</BrowserRouter>
		);
		expect(container).toMatchSnapshot();
	});
	test("should onClick correctly", async () => {
		const onClick = jest.fn();
		const label = "Path Label";
		const { getByText } = render(
			<BrowserRouter>
				<Link to="/dashboard/typography" onClick={onClick} >{label}</Link>
			</BrowserRouter>
		);
		const link = getByText(label);
		await act( () => fireEvent.click(link));

		expect(onClick).toBeCalled();
	});
	test("should has a custom className", async () => {
		const className = "custom-className";
		const label = "Path Label";
		const { getByText } = render(
			<BrowserRouter>
				<Link className={className} to="/dashboard/typography" >{label}</Link>
			</BrowserRouter>
		);
		const link = getByText(label);

		expect(link.className.includes(className)).toBeTruthy();
	});
});
