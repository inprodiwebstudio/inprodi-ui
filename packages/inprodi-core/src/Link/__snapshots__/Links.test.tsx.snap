// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links snapshot 1`] = `
<div>
  <div
    class="mantine-Text-root mantine-1mra59x"
    style="cursor: pointer; text-decoration: underline;"
  >
    Path Label
  </div>
</div>
`;
